src/main.c

```diff
@@ -112,7 +114,7 @@ static int read_exact (int fd, void *buffer, size_t buffer_size) /* {{{ */
 		status = read (fd, buffer_ptr, want_bytes);
 		if (status < 0)
 		{
-			if (errno == EAGAIN || errno == EINTR)
+			if (errno == EINTR)
```

Comment:
why this? EAGAIN can still occur, right?

Reply:
Yes but this time we want it to get out when EAGAIN occurs as it indicates a timeout has happened.
Do you know why it was being tested for before? It seems read() doesn't set errno to EAGAIN unless it's nonblocking or a timeout has been set on the socket.

Reply:
I think removing `EAGAIN` and returning instead of the right thing to do here.